github/misc.go

```diff
@@ -195,3 +196,33 @@ func (c *Client) ListServiceHooks() ([]*ServiceHook, *Response, error) {
 
 	return *hooks, resp, err
 }
+
+// ParseURL parses a GitHub URL and returns the owner and repo (if provided).
+// It returns an error if it is unable to parse the URL.
+//
+// Examples of valid URLs are:
+//     https://github.com/bazelbuild/bazel/tree/master
+//     http://github.com/bazelbuild/bazel/tree/master
+//     https://www.github.com/bazelbuild/bazel/tree/master
+//     https://github.com/bazelbuild/bazel/releases/tag/0.2.1
+//     https://github.com/bazelbuild/bazel/commit/19b5675725caf69008a717082149237400260edc
+//     https://github.com/bazelbuild/bazel/archive/0.2.1.zip
```

Comment:
It sounded like you had some specific use cases in #380 that warranted adding this helper. I don't have have a current need for this, so it's hard for me to confirm it's usefulness. However, if I did want to use this helper, the number one type of URL I'd like to use it on would be of the form:

    https://github.com/bazelbuild/bazel

Just a plain reference to a repository, and I'd want `bazelbuild`, `bazel`, `nil` as the return values. From reviewing the source code, I know that URL type would work. However, if I were a new user of this library, I think I'd be very confused by these examples, since they do not list a plain `https://github.com/bazelbuild/bazel` URL, yet the return values only include `owner` and `repo`.

So, my suggestion would be, at the very least add `https://github.com/bazelbuild/bazel` as the first example of valid URL type.